server/aesdsocket.c

```diff
@@ -0,0 +1,565 @@
+				command_status.success = false;
+				break;
+			}
+			ret = dup2(fd, STDERR_FILENO);
+			if(ret == RET_ERROR)
+			{
+				ERROR_LOG("stderr redirect failed\n");
+				syslog(LOG_ERR,"stderr redirect failed");
+				command_status.success = false;
+				break;
+			}
+			close(fd);
+		}
+		
+		/*
+		*	-> listen for connections on a socket
+		*	-> returns 0 on success, -1 on error
+		*	-> sockfd 	- socket file descriptor
+		*	-> backlog 	- No. of pending connections 
+		*			  allowed
+		*/	
+		ret = listen(socket_fd, BACKLOG_CONNECTIONS);
+		if(ret == RET_ERROR)
+		{
+			ERROR_LOG("Listen failed\n");
+			syslog(LOG_ERR,"Listen failed");
+			command_status.success = false;
+			break;
+		}
+		
+		while(1)
+		{
+			/********************************************************* 
+			*  STEP 2 :
+			*  Listens for and accepts a connection.
+			*  Logs message to the syslog 
+			*  “Accepted connection from xxx” 
+			*  where XXXX is the IP address of the connected client. 
+			*********************************************************/
+			
+			/*
+			*	-> accept a connection on a socket
+			*	-> returns a FD on success, -1 on error
+			*	-> sockfd 	- socket file descriptor
+			*	-> addr 	- location to store connection
+			*			  address of client
+			*	-> addrlen	- length of socket address
+			*/
+			accept_fd = accept(socket_fd, (struct sockaddr *)&client_addr, &client_addrlen);
+			if(accept_fd == RET_ERROR)
+			{
+				ERROR_LOG("Accept failed\n");
+				syslog(LOG_ERR,"Accept failed");
+				command_status.success = false;
+				break;
+			}
+			
+			inet_ntop(client_addr.ss_family,
+			    	   get_in_addr((struct sockaddr *)&client_addr),
+			    	   s, sizeof s);
+			//printf("server: got connection from %s\n", s);
+			syslog(LOG_INFO,"Accepted connection from %s",s);
+			
+			/********************************************************* 
+			*  STEP 3 : 
+			*  Receives data over the connection and 
+			*  appends to file /var/tmp/aesdsocketdata
+			*********************************************************/
+			
+			/*
+			*	-> receive a message from a socket
+			*	-> returns number of bytes received on success
+			*	   or -1 on error
+			*	-> sockfd 	- accepted socket file descriptor
+			*	-> buf[.len] 	- to store bytes received
+			*	-> len		- length of buffer
+			* 	-> flags	- MSG_WAITALL - Blocking mode
+			*/
+			do
+			{
+				recv_bytes = recv(accept_fd, buf, BUF_LEN, 0);
+				if(recv_bytes == RET_ERROR)
+				{
+					ERROR_LOG("Receive failed\n");
+					syslog(LOG_ERR,"Receive failed");
+					command_status.success = false;
+					break;
+				}
+				//buf[recv_bytes]='\0';
+				//DEBUG_LOG("Buffer : %ld",recv_bytes);
+				//DEBUG_LOG("Buffer : %s",buf);
+				
+				// write to file
+				ret = write(data_file_fd, buf, recv_bytes);
+				if(ret == RET_ERROR)
+				{
+					ERROR_LOG("File write failed\n");
+					syslog(LOG_ERR,"File write failed");
+					command_status.success = false;
+					break;
+				}
+			}while(buf[recv_bytes-1] != '\n');
```

Comment:
@ameyflash is there any guarantee that `recv` will receive exactly the number of bytes in a packet?  What happens when the `\n` is not at the end of the buffer?